src/Plugin/rest/resource/GetUserPreferences.php

```diff
@@ -0,0 +1,190 @@
+<?php
+namespace Drupal\ccf_user_preferences\Plugin\rest\resource;
+
+use Drupal\rest\Plugin\ResourceBase;
+use Drupal\user\Entity\User;
+use Drupal\paragraphs\Entity\Paragraph;
+use Drupal\rest\ModifiedResourceResponse;
+use Drupal\jwt\Transcoder\JwtTranscoderInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Drupal\ccf_user_preferences\Entity\CcfUserPreferences;
+
+/**
+ * Provides REST API for User preferences Details based
+ * on URL.
+ *
+ * @RestResource(
+ *   id = "get_user_preferences_rest_resource",
+ *   label = @Translation("Get User Preferences"),
+ *   uri_paths = {
+ *     "canonical" = "/api/get_user_preferences",
+ *     "create"    = "/api/add_user_preferences"
+ *   }
+ * )
+ */
+class GetUserPreferences extends ResourceBase
+{
+  /**
+   * The request object for redirect.
+   *
+   * @var \Symfony\Component\HttpFoundation\Request
+   *   The request object.
+   */
+  protected $request;
+
+  /**
+   * The JWT Transcoder service.
+   *
+   * @var \Drupal\jwt\Transcoder\JwtTranscoderInterface
+   */
+  protected $transcoder;
+
+  /**
+   * Constructor function.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param array $serializer_formats
+   *   The available serialization formats.
+   * @param \Psr\Log\LoggerInterface $logger
+   *   A logger instance.
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   The request object.
+   * @param \Drupal\jwt\Transcoder\JwtTranscoderInterface $transcoder
+   *   The JWT Transcoder service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition,
+    array $serializer_formats, LoggerInterface $logger, Request $request, JwtTranscoderInterface $transcoder) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition,$serializer_formats,$logger);
+    $this->request = $request;
+    $this->transcoder = $transcoder;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container,array
+  $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->getParameter('serializer.formats'),
+      $container->get('logger.factory')->get('rest'),
+      $container->get('request_stack')->getCurrentRequest(),
+      $container->get('jwt.transcoder')
+    );
+  }
+
+  /**
+   * Array mapping keys & fields
+   *
+   * @return void
+   */
+  public $mapper = ['tagsDietary' => 'field_dietary' ,'tagsExclusion' => 'field_exclusion','tagsMeal' => 'field_meals'];
```

Comment:
Move the class property on top , before the constructor.

Reply:
Moved the property to the top of the constructor